fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/indexes/TextIndexTest.java

```diff
@@ -1039,7 +1043,167 @@ public void saveComplexWithAggressiveConflictRanges() throws Exception {
+            recordStore.saveRecord(doc1);
+            recordStore.saveRecord(doc2);
+            commit(context);
+        }
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+            Query.InvalidExpressionException err = assertThrows(Query.InvalidExpressionException.class,
+                    () -> recordStore.deleteRecordsWhere(SIMPLE_DOC, Query.field("text").equalsValue("foo bar")));
+            assertThat(err.getMessage(), containsString("deleteRecordsWhere not supported by index SimpleDocument$text"));
+        }
+    }
+
+    @Test
+    void deleteWhereComplexByGroup() {
+        final ComplexDocument zeroGroupDocument = ComplexDocument.newBuilder()
+                .setGroup(0)
+                .setDocId(1623L)
+                .setText(TextSamples.ROMEO_AND_JULIET_PROLOGUE)
+                .build();
+        final ComplexDocument oneGroupDocument = ComplexDocument.newBuilder()
+                .setGroup(1)
+                .setDocId(1623L)
+                .setText(TextSamples.ROMEO_AND_JULIET_PROLOGUE)
+                .build();
+        final RecordMetaDataHook hook = metaDataBuilder -> {
+            TextIndexTestUtils.addRecordTypePrefix(metaDataBuilder);
+            metaDataBuilder.addIndex(COMPLEX_DOC, COMPLEX_TEXT_BY_GROUP);
+        };
+
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+            recordStore.saveRecord(zeroGroupDocument);
+            recordStore.saveRecord(oneGroupDocument);
+            context.commit();
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            openRecordStore(context, hook);
+
+            RecordQuery query = RecordQuery.newBuilder()
+                    .setRecordType(COMPLEX_DOC)
+                    .setFilter(Query.and(
+                            Query.field("group").equalsParameter("group_value"),
+                            Query.field("text").text().containsPhrase("nought could remove")
+                    ))
+                    .build();
+            RecordQueryPlan plan = recordStore.planQuery(query);
+
+            // Both groups should have a single element before delete where
+            assertThat(plan.execute(recordStore, EvaluationContext.forBinding("group_value", 0)).asList().join(), hasSize(1));
+            assertThat(plan.execute(recordStore, EvaluationContext.forBinding("group_value", 1)).asList().join(), hasSize(1));
+
+            recordStore.deleteRecordsWhere(COMPLEX_DOC, Query.field("group").equalsValue(0));
+
+            // Group zero should be gone but group 1 should still have an element remaining
+            assertThat(plan.execute(recordStore, EvaluationContext.forBinding("group_value", 0)).asList().join(), empty());
+            assertThat(plan.execute(recordStore, EvaluationContext.forBinding("group_value", 1)).asList().join(), hasSize(1));
+        }
+    }
+
+    @Test
+    void deleteWhereGroupedMap() {
```

Comment:
I believe text can have a grouped count > 1, it would probably be good to test querying that.

Reply:
Text indexes do support a `grouped count > 1` (or a `grouping count > 1`), though that's somewhat separate from the change at hand. This is mainly here because heretofore, I think there were _no_ tests of `deleteRecordsWhere` with the `TEXT` index, which seemed like a mistake. The way that the these two tests are set up, we have coverage now for when there is one grouping key and we issue a delete on that key, when there are no grouping keys, and when there are two grouping keys and we issue a delete on the first one.

In practice, you can't (currently) use the columns after the text column in a `TEXT` index, which is what the columns after the first grouped column are for. I also think, given that the grouped keys are excluded from the key expression in the `QueryToKeyMatcher`, I think the main things we need to make sure tested are:
1. If there are no grouping columns, the records-where-deleter always rejects the delete (which is tested by `deleteWhereSimpleDocuments`)
2. If the `QueryToKeyMatcher` finds a match in the grouping keys, that the index data actually gets deleted (which is tested by the other two tests)